docs/src/imagevolume_extensions.md

```diff
@@ -0,0 +1,332 @@
+
+- `extension_control_path` to `/extensions/<EXTENSION_NAME>/share`, allowing
+  PostgreSQL to locate any extension control file within `/extensions/<EXTENSION_NAME>/share/extension`
+- `dynamic_library_path` to `/extensions/<EXTENSION_NAME>/lib`
+
+These values are appended in the order in which the extensions are defined in
+the `extensions` list, ensuring deterministic path resolution within
+PostgreSQL. This allows PostgreSQL to discover and load the extension without
+requiring manual configuration inside the pod.
+
+!!! Info
+    Depending on how your extension container images are built and their layout,
+    you may need to adjust the default `extension_control_path` and
+    `dynamic_library_path` values to match the image structure.
+
+!!! Important
+    The extension container image must match the PostgreSQL container used by
+    your cluster in PostgreSQL major version, Operating system distribution, and
+    CPU architecture to ensure compatibility and prevent runtime issues.
+
+## How to add a new extension
+
+Adding an extension to a database in CloudNativePG involves two steps:
+
+1. Attach the extension image to the `Cluster` resource so that PostgreSQL can
+   discover and load it.
+2. Declare the extension in the `Database` resource where you want it
+   installed.
+
+For illustration, we will use a simple, fictitious extension named `foo`.
+
+### Adding a new extension to a `Cluster` resource
+
+You can add an `ImageVolume`-based extension to a `Cluster` using the
+`.spec.postgresql.extensions` stanza. For example:
+
+```yaml
+apiVersion: postgresql.cnpg.io/v1
+kind: Cluster
+metadata:
+  name: foo-18
+spec:
+  # ...
+  postgresql:
+    extensions:
+      - name: foo
+        image:
+          reference: # registry path for your extension image
+  # ...
+```
+
+The `name` field is **mandatory** and **must be unique within the cluster**, as
+it determines the mount path (`/extensions/foo` in this example). It must
+consist of *lowercase alphanumeric characters or hyphens (`-`)* and must start
+and end with an alphanumeric character.
+
+The `image` stanza follows the [Kubernetes `ImageVolume` API](https://kubernetes.io/docs/tasks/configure-pod-container/image-volumes/).
+The `reference` must point to a valid container registry path for the extension
+image.
+
+!!! Important
+    When a new extension is added to a running `Cluster`, CloudNativePG will
+    automatically trigger a [rolling update](rolling_update.md) to attach the new
+    image volume to each pod. Before adding a new extension in production,
+    ensure you have thoroughly tested it in a staging environment to prevent
+    configuration issues that could leave your PostgreSQL cluster in an unhealthy
+    state.
+
+Once mounted, CloudNativePG will automatically configure PostgreSQL by appending:
+
+- `/extensions/foo/share` to `extension_control_path`
+- `/extensions/foo/lib` to `dynamic_library_path`
```

Comment:
Postgres will rewrite those as

```
/extension/foo/share/postgreql/share
/extension/foo/share/postgreql/lib
```

unless the default path is updated to contain the string "pgsql" or "postgresql".

Reply:
That too!